consensus/src/quorum_store/batch_generator.rs

```diff
@@ -129,49 +163,50 @@ impl BatchGenerator {
                 .observe_duration(Duration::from_secs_f64(duration));
 
             self.last_end_batch_time = Instant::now();
-            return None;
+            return vec![];
         } else {
             counters::PULLED_TXNS_COUNT.inc();
             counters::PULLED_TXNS_NUM.observe(pulled_txns.len() as f64);
         }
 
         // Quorum store metrics
-        counters::CREATED_BATCHES_COUNT.inc();
-
         let duration = self.last_end_batch_time.elapsed().as_secs_f64();
         counters::BATCH_CREATION_DURATION.observe_duration(Duration::from_secs_f64(duration));
 
-        counters::NUM_TXN_PER_BATCH.observe(pulled_txns.len() as f64);
-
-        let batch_id = self.batch_id;
-        self.batch_id.increment();
-        self.db
-            .save_batch_id(self.epoch, self.batch_id)
-            .expect("Could not save to db");
-
         let expiry_time = aptos_infallible::duration_since_epoch().as_micros() as u64
             + self.config.batch_expiry_gap_when_init_usecs;
-        let txn_summaries: Vec<_> = pulled_txns
-            .iter()
-            .map(|txn| TransactionSummary {
-                sender: txn.sender(),
-                sequence_number: txn.sequence_number(),
-            })
-            .collect();
 
-        let batch = Batch::new(
-            batch_id,
-            pulled_txns,
-            self.epoch,
-            expiry_time,
-            self.my_peer_id,
-        );
+        pulled_txns.sort_by_key(|txn| txn.gas_unit_price());
```

Comment:
so we don't take seqnum into considerations even from the same pull?

Reply:
Yes, in this PR we don't consider the sequence number at all. A future PR (or update on this PR) will consider both the current pull and the excluded txns.

Reply:
I added it to this PR: 5bcb5784e209da8df83c3408a0c2292a46189472